omnigibson/utils/usd_utils.py

```diff
@@ -1032,8 +1018,188 @@ def get_relative_jacobian(self, prim_path):
         return tf @ jacobian
 
 
-ControllableObjectViewAPI = BatchControlViewAPIImpl("/World/scene_*/controllable_*/base_link")
-DummyControllableObjectViewAPI = BatchControlViewAPIImpl("/World/scene_*/dummy_*/base_link")
+class ControllableObjectViewAPI:
+    """
+    An interface that creates BatchControlViewAPIImpl instances for each robot type in the scene.
+
+    This is done to avoid the overhead of reading from many views for each robot in each physics step,
+    providing major speed improvements in vector env use cases.
+
+    This class is a singleton, and should be used to access the BatchControlViewAPIImpl instances.
+
+    The pattern used to group the robots is based on the robot prim paths, which is assumed to be in the format
+    /World/scene_*/controllable__robottype__robotname or /World/scene_*/dummy__robottype__robotname.
+
+    The patterns used by the subviews are generated by replacing the robot name with a wildcard, so that all robots
+    of the same type are grouped together. If there are fixed base robots, they will be grouped separately from
+    non-fixed base robots even within the same robot type, by virtue of their different articulation root paths.
+    """
+
+    @classmethod
+    def clear(cls):
+        for view in cls._view_by_pattern.values():
+            view.clear()
+
+    @classmethod
+    def flush_control(cls):
+        for view in cls._view_by_pattern.values():
+            view.flush_control()
+
+    @classmethod
+    def initialize_view(cls):
+        cls._view_by_pattern = {}
+
+        # First, get all of the controllable objects in the scene (avoiding circular import)
+        from omnigibson.objects.controllable_object import ControllableObject
+
+        controllable_objects = [
+            obj for scene in og.sim.scenes for obj in scene.objects if isinstance(obj, ControllableObject)
+        ]
+
+        # Get their corresponding prim paths
+        expected_regular_prim_paths = {obj.articulation_root_path for obj in controllable_objects}
+        expected_dummy_prim_paths = {
+            obj._dummy.articulation_root_path
```

Comment:
wish there was a more elegant way to infer this but for now this is probably ok

Reply:
Yeah like a "get_controllable_prim_paths" or something but it'd be very specific. I think we can live with this for now.